lib/matplotlib/ticker.py

```diff
@@ -377,19 +380,39 @@ class FuncFormatter(Formatter):
     """
     Use a user-defined function for formatting.
 
-    The function should take in two inputs (a tick value ``x`` and a
-    position ``pos``), and return a string containing the corresponding
-    tick label.
+    The function can take in at most two inputs (a required tick value ``x``
+     and an optional position ``pos``), and must return a string containing
+     the corresponding tick label.
     """
     def __init__(self, func):
         self.func = func
 
+        if not isinstance(func, types.BuiltinFunctionType):
```

Comment:
`func` is part of the public API, which means it can be changed at runtime.  So you can't count on the number of arguments being the same when this is actually called.  `func` probably needs to be changed into a property which recalculates the number of arguments in the setter.